### PR TITLE
Add ResNet 50 training code

### PR DESCRIPTION
**Pull request by jenniew**
Add ResNet50 Training code for imagenet using FeatureSet and Estimator. 
Use "--memoryType PMEM" to cache with AEP.
Add mkldnn support.

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala [hunk 1]
```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2016 The BigDL Authors.
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala]: zoo copyright？

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        (224, DatasetType.ImageNet, param.nepochs)
+
+      val memType = param.memoryType match {
+        case "dram" =>
+          DRAM
+        case "pmem" =>
+          logger.info("~~~~~~~ Caching with AEP ~~~~~~~")
+          PMEM
+        case "direct" =>
+          logger.info("~~~~~~~ Caching with DIRECT ~~~~~~~")
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala]: Delete this, we don't expose Direct... Just for internal test.

Comment (zhichao-li) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala]: I don't think you need any of this. You should refer to `inception/Train.scala` and use `MemoryType.fromString(param.memoryType)` instead.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala (hunk 1 above)
Comment (jenniew) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala]: updated

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala (hunk 2 above)
Comment (jenniew) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/resnet/TrainImageNet.scala]: changed

**Pull request merged**